{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c77f8da1-2837-4bf7-b3f4-dfcaa7ab42f6",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-11-08T21:37:44.240180Z",
     "iopub.status.busy": "2023-11-08T21:37:44.240054Z",
     "iopub.status.idle": "2023-11-08T21:37:44.324130Z",
     "shell.execute_reply": "2023-11-08T21:37:44.323507Z"
    }
   },
   "outputs": [],
   "source": [
    "import scipy.io\n",
    "import numpy as np\n",
    "import random\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3fe25376-b856-4296-ba02-5e3771ff9451",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-11-08T21:37:44.326506Z",
     "iopub.status.busy": "2023-11-08T21:37:44.326344Z",
     "iopub.status.idle": "2023-11-08T23:26:20.905786Z",
     "shell.execute_reply": "2023-11-08T23:26:20.905112Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 16:37:44.509180: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
      "To enable the following instructions: FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "/usr/lib/python3.11/site-packages/h5py/__init__.py:36: UserWarning: h5py is running against HDF5 1.14.2 when it was built against 1.14.1, this may cause problems\n",
      "  _warn((\"h5py is running against HDF5 {0} when it was built against {1}, \"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "35057\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/compat/v2_compat.py:107: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "non-resource variables are not supported in the long term\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 16:39:09.301593: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:09.312999: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:09.313233: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:09.316486: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:09.316674: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:09.316844: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:39.439586: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:39.440291: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:39.440489: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:39.440644: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1639] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 46608 MB memory:  -> device: 0, name: NVIDIA RTX A6000, pci bus id: 0000:2d:00.0, compute capability: 8.6\n",
      "2023-11-08 16:39:39.441002: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 16:39:40.351536: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:40.351782: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:40.351958: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:40.352285: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:40.352468: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:40.352605: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1639] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 46608 MB memory:  -> device: 0, name: NVIDIA RTX A6000, pci bus id: 0000:2d:00.0, compute capability: 8.6\n",
      "2023-11-08 16:39:40.352636: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.\n",
      "2023-11-08 16:39:40.374835: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:375] MLIR V1 optimization pass is not enabled\n",
      "2023-11-08 16:39:40.633089: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_21/lstm_cell_21/bias/Assign' id:3510 op device:{requested: '', assigned: ''} def:{{{node lstm_21/lstm_cell_21/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_21/lstm_cell_21/bias, lstm_21/lstm_cell_21/bias/Initializer/concat)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 16:39:40.766669: W tensorflow/c/c_api.cc:304] Operation '{name:'strided_slice/stack_1' id:3634 op device:{requested: '', assigned: ''} def:{{{node strided_slice/stack_1}} = Const[_has_manual_control_dependencies=true, dtype=DT_INT32, value=Tensor<type: int32 shape: [4] values: 0 0 0...>]()}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 16:39:40.800485: W tensorflow/c/c_api.cc:304] Operation '{name:'strided_slice/stack_2' id:3635 op device:{requested: '', assigned: ''} def:{{{node strided_slice/stack_2}} = Const[_has_manual_control_dependencies=true, dtype=DT_INT32, value=Tensor<type: int32 shape: [4] values: 1 1 1...>]()}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "(31543, 95)\n",
      "Train on 31543 samples, validate on 3514 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 16:39:45.207936: W tensorflow/c/c_api.cc:304] Operation '{name:'training/Adam/conv2d_1/bias/v/Assign' id:17012 op device:{requested: '', assigned: ''} def:{{{node training/Adam/conv2d_1/bias/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](training/Adam/conv2d_1/bias/v, training/Adam/conv2d_1/bias/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 16:39:49.539972: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n",
      "2023-11-08 16:39:52.125503: I tensorflow/compiler/xla/stream_executor/cuda/cuda_blas.cc:606] TensorFloat-32 will be used for the matrix multiplication. This will only be logged once.\n",
      "2023-11-08 16:39:52.134074: I tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:432] Loaded cuDNN version 8902\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31543/31543 [==============================] - ETA: 0s - loss: 3.1647"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/lib/python3.11/site-packages/keras/engine/training_v1.py:2335: UserWarning: `Model.state_updates` will be removed in a future version. This property should not be used in TensorFlow 2.0, as `updates` are applied automatically.\n",
      "  updates = self.state_updates\n",
      "2023-11-08 16:40:13.001745: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_1/mul' id:6477 op device:{requested: '', assigned: ''} def:{{{node loss_1/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_1/mul/x, loss_1/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.77886, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 29s 907us/sample - loss: 3.1647 - val_loss: 1.7789\n",
      "Epoch 2/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.7873\n",
      "Epoch 2: val_loss improved from 1.77886 to 1.56590, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 21s 664us/sample - loss: 1.7873 - val_loss: 1.5659\n",
      "Epoch 3/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.6187\n",
      "Epoch 3: val_loss improved from 1.56590 to 1.49973, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 22s 700us/sample - loss: 1.6187 - val_loss: 1.4997\n",
      "Epoch 4/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.5620\n",
      "Epoch 4: val_loss improved from 1.49973 to 1.47088, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 20s 630us/sample - loss: 1.5620 - val_loss: 1.4709\n",
      "Epoch 5/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.5349\n",
      "Epoch 5: val_loss improved from 1.47088 to 1.45119, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 20s 618us/sample - loss: 1.5349 - val_loss: 1.4512\n",
      "Epoch 6/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.5181\n",
      "Epoch 6: val_loss improved from 1.45119 to 1.43978, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 20s 623us/sample - loss: 1.5181 - val_loss: 1.4398\n",
      "Epoch 7/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.5070\n",
      "Epoch 7: val_loss improved from 1.43978 to 1.43464, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 20s 622us/sample - loss: 1.5070 - val_loss: 1.4346\n",
      "Epoch 8/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.5020\n",
      "Epoch 8: val_loss improved from 1.43464 to 1.41938, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 19s 618us/sample - loss: 1.5020 - val_loss: 1.4194\n",
      "Epoch 9/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4974\n",
      "Epoch 9: val_loss improved from 1.41938 to 1.41715, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 20s 622us/sample - loss: 1.4974 - val_loss: 1.4171\n",
      "Epoch 10/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.5284\n",
      "Epoch 10: val_loss did not improve from 1.41715\n",
      "31543/31543 [==============================] - 21s 664us/sample - loss: 1.5284 - val_loss: 1.4180\n",
      "Epoch 11/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4910\n",
      "Epoch 11: val_loss improved from 1.41715 to 1.41114, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 22s 702us/sample - loss: 1.4910 - val_loss: 1.4111\n",
      "Epoch 12/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4820\n",
      "Epoch 12: val_loss improved from 1.41114 to 1.40198, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 22s 702us/sample - loss: 1.4820 - val_loss: 1.4020\n",
      "Epoch 13/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4916\n",
      "Epoch 13: val_loss improved from 1.40198 to 1.40093, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 20s 628us/sample - loss: 1.4916 - val_loss: 1.4009\n",
      "Epoch 14/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4814\n",
      "Epoch 14: val_loss improved from 1.40093 to 1.39836, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 22s 706us/sample - loss: 1.4814 - val_loss: 1.3984\n",
      "Epoch 15/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4896\n",
      "Epoch 15: val_loss improved from 1.39836 to 1.39565, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 22s 698us/sample - loss: 1.4896 - val_loss: 1.3957\n",
      "Epoch 16/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4688\n",
      "Epoch 16: val_loss improved from 1.39565 to 1.39275, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 22s 695us/sample - loss: 1.4688 - val_loss: 1.3927\n",
      "Epoch 17/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4635\n",
      "Epoch 17: val_loss improved from 1.39275 to 1.38794, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 22s 693us/sample - loss: 1.4635 - val_loss: 1.3879\n",
      "Epoch 18/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4745\n",
      "Epoch 18: val_loss improved from 1.38794 to 1.38607, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 22s 692us/sample - loss: 1.4745 - val_loss: 1.3861\n",
      "Epoch 19/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4695\n",
      "Epoch 19: val_loss improved from 1.38607 to 1.38570, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 21s 651us/sample - loss: 1.4695 - val_loss: 1.3857\n",
      "Epoch 20/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.5276\n",
      "Epoch 20: val_loss did not improve from 1.38570\n",
      "31543/31543 [==============================] - 20s 621us/sample - loss: 1.5276 - val_loss: 1.4050\n",
      "Epoch 21/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.5380\n",
      "Epoch 21: val_loss did not improve from 1.38570\n",
      "31543/31543 [==============================] - 20s 622us/sample - loss: 1.5380 - val_loss: 1.4232\n",
      "Epoch 22/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4787\n",
      "Epoch 22: val_loss did not improve from 1.38570\n",
      "31543/31543 [==============================] - 20s 625us/sample - loss: 1.4787 - val_loss: 1.4016\n",
      "Epoch 23/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4601\n",
      "Epoch 23: val_loss did not improve from 1.38570\n",
      "31543/31543 [==============================] - 19s 615us/sample - loss: 1.4601 - val_loss: 1.3904\n",
      "Epoch 24/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4524\n",
      "Epoch 24: val_loss did not improve from 1.38570\n",
      "31543/31543 [==============================] - 20s 624us/sample - loss: 1.4524 - val_loss: 1.3894\n",
      "Epoch 25/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4471\n",
      "Epoch 25: val_loss did not improve from 1.38570\n",
      "31543/31543 [==============================] - 21s 652us/sample - loss: 1.4471 - val_loss: 1.3859\n",
      "Epoch 26/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4410\n",
      "Epoch 26: val_loss improved from 1.38570 to 1.38142, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 22s 688us/sample - loss: 1.4410 - val_loss: 1.3814\n",
      "Epoch 27/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4383\n",
      "Epoch 27: val_loss improved from 1.38142 to 1.37948, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 22s 705us/sample - loss: 1.4383 - val_loss: 1.3795\n",
      "Epoch 28/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4366\n",
      "Epoch 28: val_loss did not improve from 1.37948\n",
      "31543/31543 [==============================] - 21s 673us/sample - loss: 1.4366 - val_loss: 1.3804\n",
      "Epoch 29/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4340\n",
      "Epoch 29: val_loss did not improve from 1.37948\n",
      "31543/31543 [==============================] - 20s 619us/sample - loss: 1.4340 - val_loss: 1.3850\n",
      "Epoch 30/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4320\n",
      "Epoch 30: val_loss improved from 1.37948 to 1.37731, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 20s 641us/sample - loss: 1.4320 - val_loss: 1.3773\n",
      "Epoch 31/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4292\n",
      "Epoch 31: val_loss improved from 1.37731 to 1.37256, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 20s 618us/sample - loss: 1.4292 - val_loss: 1.3726\n",
      "Epoch 32/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4275\n",
      "Epoch 32: val_loss did not improve from 1.37256\n",
      "31543/31543 [==============================] - 20s 648us/sample - loss: 1.4275 - val_loss: 1.3771\n",
      "Epoch 33/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4251\n",
      "Epoch 33: val_loss did not improve from 1.37256\n",
      "31543/31543 [==============================] - 20s 650us/sample - loss: 1.4251 - val_loss: 1.3744\n",
      "Epoch 34/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4223\n",
      "Epoch 34: val_loss improved from 1.37256 to 1.37137, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 21s 654us/sample - loss: 1.4223 - val_loss: 1.3714\n",
      "Epoch 35/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4203\n",
      "Epoch 35: val_loss did not improve from 1.37137\n",
      "31543/31543 [==============================] - 22s 704us/sample - loss: 1.4203 - val_loss: 1.3735\n",
      "Epoch 36/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4178\n",
      "Epoch 36: val_loss improved from 1.37137 to 1.36628, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 22s 709us/sample - loss: 1.4178 - val_loss: 1.3663\n",
      "Epoch 37/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4182\n",
      "Epoch 37: val_loss did not improve from 1.36628\n",
      "31543/31543 [==============================] - 21s 678us/sample - loss: 1.4182 - val_loss: 1.3704\n",
      "Epoch 38/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4130\n",
      "Epoch 38: val_loss did not improve from 1.36628\n",
      "31543/31543 [==============================] - 20s 620us/sample - loss: 1.4130 - val_loss: 1.3685\n",
      "Epoch 39/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4106\n",
      "Epoch 39: val_loss improved from 1.36628 to 1.36254, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 19s 603us/sample - loss: 1.4106 - val_loss: 1.3625\n",
      "Epoch 40/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4113\n",
      "Epoch 40: val_loss improved from 1.36254 to 1.36093, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 19s 604us/sample - loss: 1.4113 - val_loss: 1.3609\n",
      "Epoch 41/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4072\n",
      "Epoch 41: val_loss improved from 1.36093 to 1.36083, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 19s 607us/sample - loss: 1.4072 - val_loss: 1.3608\n",
      "Epoch 42/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4089\n",
      "Epoch 42: val_loss improved from 1.36083 to 1.35856, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 19s 607us/sample - loss: 1.4089 - val_loss: 1.3586\n",
      "Epoch 43/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4062\n",
      "Epoch 43: val_loss did not improve from 1.35856\n",
      "31543/31543 [==============================] - 19s 595us/sample - loss: 1.4062 - val_loss: 1.3594\n",
      "Epoch 44/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4068\n",
      "Epoch 44: val_loss improved from 1.35856 to 1.35729, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 19s 596us/sample - loss: 1.4068 - val_loss: 1.3573\n",
      "Epoch 45/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4037\n",
      "Epoch 45: val_loss did not improve from 1.35729\n",
      "31543/31543 [==============================] - 22s 696us/sample - loss: 1.4037 - val_loss: 1.3633\n",
      "Epoch 46/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4019\n",
      "Epoch 46: val_loss improved from 1.35729 to 1.35216, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 21s 656us/sample - loss: 1.4019 - val_loss: 1.3522\n",
      "Epoch 47/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3992\n",
      "Epoch 47: val_loss improved from 1.35216 to 1.35119, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 20s 646us/sample - loss: 1.3992 - val_loss: 1.3512\n",
      "Epoch 48/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3972\n",
      "Epoch 48: val_loss did not improve from 1.35119\n",
      "31543/31543 [==============================] - 21s 670us/sample - loss: 1.3972 - val_loss: 1.3576\n",
      "Epoch 49/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3957\n",
      "Epoch 49: val_loss did not improve from 1.35119\n",
      "31543/31543 [==============================] - 19s 599us/sample - loss: 1.3957 - val_loss: 1.3522\n",
      "Epoch 50/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3933\n",
      "Epoch 50: val_loss improved from 1.35119 to 1.34704, saving model to ./checkpoints/unknown_person_baseline_p12_20.h5\n",
      "31543/31543 [==============================] - 19s 608us/sample - loss: 1.3933 - val_loss: 1.3470\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling GlorotUniform.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling Zeros.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling Orthogonal.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 16:57:01.760768: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_22_1/lstm_cell_59/kernel/Assign' id:22162 op device:{requested: '', assigned: ''} def:{{{node lstm_22_1/lstm_cell_59/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_22_1/lstm_cell_59/kernel, lstm_22_1/lstm_cell_59/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 16:57:03.442354: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_33_1/lstm_cell_70/recurrent_kernel/v/Assign' id:26395 op device:{requested: '', assigned: ''} def:{{{node lstm_33_1/lstm_cell_70/recurrent_kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_33_1/lstm_cell_70/recurrent_kernel/v, lstm_33_1/lstm_cell_70/recurrent_kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "/usr/lib/python3.11/site-packages/keras/engine/training_v1.py:2359: UserWarning: `Model.state_updates` will be removed in a future version. This property should not be used in TensorFlow 2.0, as `updates` are applied automatically.\n",
      "  updates=self.state_updates,\n",
      "2023-11-08 16:57:04.738898: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_1/cond/Merge' id:24591 op device:{requested: '', assigned: ''} def:{{{node dropout_4_1/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_1/cond/Identity, dropout_4_1/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1944)\n",
      "(1514, 1944)\n",
      "(1644, 1944)\n",
      "(1764, 1944)\n",
      "(1836, 1944)\n",
      "(1699, 1944)\n",
      "(1369, 1944)\n",
      "(1766, 1944)\n",
      "(1631, 1944)\n",
      "(1538, 1944)\n",
      "(1932, 1944)\n",
      "(1679, 1944)\n",
      "(1776, 1944)\n",
      "(1872, 1944)\n",
      "(1692, 1944)\n",
      "(1824, 1944)\n",
      "(970, 1944)\n",
      "(1656, 1944)\n",
      "(1896, 1944)\n",
      "{1: 6.126542411249586, 2: 5.266886674721579, 4: 9.525109787634438, 5: 5.488345021102138, 6: 5.03745625678774, 8: 8.904275264139502, 9: 6.097285362310649, 10: 7.86060888412994, 11: 6.484823721155447, 13: 6.778344150438463, 17: 10.0, 19: 8.342932273058459, 21: 9.222623355476703, 22: 1.0, 25: 7.808491414797922, 26: 6.191594735081483, 27: 4.280703343501259, 28: 6.269562667163607, 29: 1.7183937629931278}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2792763/2605367892.py:300: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)\n",
      "  w_train[i] = weights_dict[int(p_train[i])]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31543 samples, validate on 3514 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 17:00:47.490927: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_5/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31543/31543 [==============================] - ETA: 0s - loss: 9.5062\n",
      "Epoch 1: val_loss improved from inf to 1.36683, saving model to ./checkpoints/unknown_person_p12_20.h5\n",
      "31543/31543 [==============================] - 25s 801us/sample - loss: 9.5062 - val_loss: 1.3668\n",
      "Epoch 2/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.4572\n",
      "Epoch 2: val_loss improved from 1.36683 to 1.36247, saving model to ./checkpoints/unknown_person_p12_20.h5\n",
      "31543/31543 [==============================] - 19s 615us/sample - loss: 9.4572 - val_loss: 1.3625\n",
      "Epoch 3/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.3943\n",
      "Epoch 3: val_loss did not improve from 1.36247\n",
      "31543/31543 [==============================] - 19s 612us/sample - loss: 9.3943 - val_loss: 1.3655\n",
      "Epoch 4/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.3752\n",
      "Epoch 4: val_loss improved from 1.36247 to 1.35947, saving model to ./checkpoints/unknown_person_p12_20.h5\n",
      "31543/31543 [==============================] - 19s 615us/sample - loss: 9.3752 - val_loss: 1.3595\n",
      "Epoch 5/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.3307\n",
      "Epoch 5: val_loss improved from 1.35947 to 1.35725, saving model to ./checkpoints/unknown_person_p12_20.h5\n",
      "31543/31543 [==============================] - 20s 624us/sample - loss: 9.3307 - val_loss: 1.3573\n",
      "Epoch 6/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.3203\n",
      "Epoch 6: val_loss did not improve from 1.35725\n",
      "31543/31543 [==============================] - 21s 668us/sample - loss: 9.3203 - val_loss: 1.3585\n",
      "Epoch 7/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.2791\n",
      "Epoch 7: val_loss improved from 1.35725 to 1.35682, saving model to ./checkpoints/unknown_person_p12_20.h5\n",
      "31543/31543 [==============================] - 19s 611us/sample - loss: 9.2791 - val_loss: 1.3568\n",
      "Epoch 8/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.2667\n",
      "Epoch 8: val_loss did not improve from 1.35682\n",
      "31543/31543 [==============================] - 19s 603us/sample - loss: 9.2667 - val_loss: 1.3692\n",
      "Epoch 9/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.2658\n",
      "Epoch 9: val_loss improved from 1.35682 to 1.34929, saving model to ./checkpoints/unknown_person_p12_20.h5\n",
      "31543/31543 [==============================] - 20s 644us/sample - loss: 9.2658 - val_loss: 1.3493\n",
      "Epoch 10/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.2653\n",
      "Epoch 10: val_loss did not improve from 1.34929\n",
      "31543/31543 [==============================] - 20s 645us/sample - loss: 9.2653 - val_loss: 1.3619\n",
      "Epoch 11/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.2511\n",
      "Epoch 11: val_loss improved from 1.34929 to 1.34796, saving model to ./checkpoints/unknown_person_p12_20.h5\n",
      "31543/31543 [==============================] - 21s 652us/sample - loss: 9.2511 - val_loss: 1.3480\n",
      "Epoch 12/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.2416\n",
      "Epoch 12: val_loss did not improve from 1.34796\n",
      "31543/31543 [==============================] - 22s 683us/sample - loss: 9.2416 - val_loss: 1.3511\n",
      "Epoch 13/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.2153\n",
      "Epoch 13: val_loss improved from 1.34796 to 1.34576, saving model to ./checkpoints/unknown_person_p12_20.h5\n",
      "31543/31543 [==============================] - 22s 689us/sample - loss: 9.2153 - val_loss: 1.3458\n",
      "Epoch 14/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.1961\n",
      "Epoch 14: val_loss did not improve from 1.34576\n",
      "31543/31543 [==============================] - 22s 683us/sample - loss: 9.1961 - val_loss: 1.3466\n",
      "Epoch 15/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.1770\n",
      "Epoch 15: val_loss did not improve from 1.34576\n",
      "31543/31543 [==============================] - 22s 686us/sample - loss: 9.1770 - val_loss: 1.3465\n",
      "Epoch 16/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.1707\n",
      "Epoch 16: val_loss did not improve from 1.34576\n",
      "31543/31543 [==============================] - 20s 634us/sample - loss: 9.1707 - val_loss: 1.3473\n",
      "Epoch 17/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.1743\n",
      "Epoch 17: val_loss did not improve from 1.34576\n",
      "31543/31543 [==============================] - 18s 582us/sample - loss: 9.1743 - val_loss: 1.3474\n",
      "Epoch 18/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.1716\n",
      "Epoch 18: val_loss did not improve from 1.34576\n",
      "31543/31543 [==============================] - 19s 598us/sample - loss: 9.1716 - val_loss: 1.3479\n",
      "Epoch 19/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.1534\n",
      "Epoch 19: val_loss did not improve from 1.34576\n",
      "31543/31543 [==============================] - 18s 586us/sample - loss: 9.1534 - val_loss: 1.3501\n",
      "Epoch 20/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.1302\n",
      "Epoch 20: val_loss improved from 1.34576 to 1.34492, saving model to ./checkpoints/unknown_person_p12_20.h5\n",
      "31543/31543 [==============================] - 19s 589us/sample - loss: 9.1302 - val_loss: 1.3449\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 17:07:33.682211: W tensorflow/c/c_api.cc:304] Operation '{name:'dense_5/kernel/Assign' id:43960 op device:{requested: '', assigned: ''} def:{{{node dense_5/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](dense_5/kernel, dense_5/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 17:07:36.035863: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_20_2/lstm_cell_94/kernel/m/Assign' id:44950 op device:{requested: '', assigned: ''} def:{{{node lstm_20_2/lstm_cell_94/kernel/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_20_2/lstm_cell_94/kernel/m, lstm_20_2/lstm_cell_94/kernel/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31543 samples, validate on 3514 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 17:07:41.047618: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_5/mul' id:44579 op device:{requested: '', assigned: ''} def:{{{node loss_5/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_5/mul/x, loss_5/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 17:07:53.444106: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_6/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3936"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 17:08:14.905464: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_5/mul' id:44579 op device:{requested: '', assigned: ''} def:{{{node loss_5/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_5/mul/x, loss_5/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.35733, saving model to ./checkpoints/unknown_person_baseline_2_p12_20.h5\n",
      "31543/31543 [==============================] - 27s 871us/sample - loss: 1.3936 - val_loss: 1.3573\n",
      "Epoch 2/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3913\n",
      "Epoch 2: val_loss improved from 1.35733 to 1.35314, saving model to ./checkpoints/unknown_person_baseline_2_p12_20.h5\n",
      "31543/31543 [==============================] - 19s 617us/sample - loss: 1.3913 - val_loss: 1.3531\n",
      "Epoch 3/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3886\n",
      "Epoch 3: val_loss improved from 1.35314 to 1.35209, saving model to ./checkpoints/unknown_person_baseline_2_p12_20.h5\n",
      "31543/31543 [==============================] - 20s 647us/sample - loss: 1.3886 - val_loss: 1.3521\n",
      "Epoch 4/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3919\n",
      "Epoch 4: val_loss improved from 1.35209 to 1.35103, saving model to ./checkpoints/unknown_person_baseline_2_p12_20.h5\n",
      "31543/31543 [==============================] - 22s 684us/sample - loss: 1.3919 - val_loss: 1.3510\n",
      "Epoch 5/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3891\n",
      "Epoch 5: val_loss improved from 1.35103 to 1.35022, saving model to ./checkpoints/unknown_person_baseline_2_p12_20.h5\n",
      "31543/31543 [==============================] - 22s 689us/sample - loss: 1.3891 - val_loss: 1.3502\n",
      "Epoch 6/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3865\n",
      "Epoch 6: val_loss improved from 1.35022 to 1.34266, saving model to ./checkpoints/unknown_person_baseline_2_p12_20.h5\n",
      "31543/31543 [==============================] - 22s 698us/sample - loss: 1.3865 - val_loss: 1.3427\n",
      "Epoch 7/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3855\n",
      "Epoch 7: val_loss did not improve from 1.34266\n",
      "31543/31543 [==============================] - 22s 697us/sample - loss: 1.3855 - val_loss: 1.3436\n",
      "Epoch 8/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3850\n",
      "Epoch 8: val_loss did not improve from 1.34266\n",
      "31543/31543 [==============================] - 22s 693us/sample - loss: 1.3850 - val_loss: 1.3441\n",
      "Epoch 9/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3836\n",
      "Epoch 9: val_loss did not improve from 1.34266\n",
      "31543/31543 [==============================] - 22s 693us/sample - loss: 1.3836 - val_loss: 1.3523\n",
      "Epoch 10/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3825\n",
      "Epoch 10: val_loss did not improve from 1.34266\n",
      "31543/31543 [==============================] - 22s 689us/sample - loss: 1.3825 - val_loss: 1.3448\n",
      "Epoch 11/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3834\n",
      "Epoch 11: val_loss did not improve from 1.34266\n",
      "31543/31543 [==============================] - 21s 674us/sample - loss: 1.3834 - val_loss: 1.3460\n",
      "Epoch 12/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3791\n",
      "Epoch 12: val_loss did not improve from 1.34266\n",
      "31543/31543 [==============================] - 19s 602us/sample - loss: 1.3791 - val_loss: 1.3466\n",
      "Epoch 13/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3782\n",
      "Epoch 13: val_loss improved from 1.34266 to 1.34134, saving model to ./checkpoints/unknown_person_baseline_2_p12_20.h5\n",
      "31543/31543 [==============================] - 19s 610us/sample - loss: 1.3782 - val_loss: 1.3413\n",
      "Epoch 14/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3765\n",
      "Epoch 14: val_loss improved from 1.34134 to 1.33673, saving model to ./checkpoints/unknown_person_baseline_2_p12_20.h5\n",
      "31543/31543 [==============================] - 20s 637us/sample - loss: 1.3765 - val_loss: 1.3367\n",
      "Epoch 15/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3770\n",
      "Epoch 15: val_loss did not improve from 1.33673\n",
      "31543/31543 [==============================] - 21s 670us/sample - loss: 1.3770 - val_loss: 1.3390\n",
      "Epoch 16/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3735\n",
      "Epoch 16: val_loss did not improve from 1.33673\n",
      "31543/31543 [==============================] - 19s 606us/sample - loss: 1.3735 - val_loss: 1.3398\n",
      "Epoch 17/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3739\n",
      "Epoch 17: val_loss did not improve from 1.33673\n",
      "31543/31543 [==============================] - 19s 610us/sample - loss: 1.3739 - val_loss: 1.3389\n",
      "Epoch 18/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3726\n",
      "Epoch 18: val_loss did not improve from 1.33673\n",
      "31543/31543 [==============================] - 21s 652us/sample - loss: 1.3726 - val_loss: 1.3370\n",
      "Epoch 19/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3713\n",
      "Epoch 19: val_loss did not improve from 1.33673\n",
      "31543/31543 [==============================] - 21s 663us/sample - loss: 1.3713 - val_loss: 1.3391\n",
      "Epoch 20/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3698\n",
      "Epoch 20: val_loss did not improve from 1.33673\n",
      "31543/31543 [==============================] - 19s 610us/sample - loss: 1.3698 - val_loss: 1.3407\n",
      "Train on 31543 samples, validate on 3514 samples\n",
      "Epoch 1/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3674\n",
      "Epoch 1: val_loss improved from inf to 1.33749, saving model to ./checkpoints/unknown_person_baseline_p12_21.h5\n",
      "31543/31543 [==============================] - 19s 612us/sample - loss: 1.3674 - val_loss: 1.3375\n",
      "Epoch 2/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3688\n",
      "Epoch 2: val_loss did not improve from 1.33749\n",
      "31543/31543 [==============================] - 19s 614us/sample - loss: 1.3688 - val_loss: 1.3417\n",
      "Epoch 3/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3674\n",
      "Epoch 3: val_loss improved from 1.33749 to 1.33482, saving model to ./checkpoints/unknown_person_baseline_p12_21.h5\n",
      "31543/31543 [==============================] - 19s 618us/sample - loss: 1.3674 - val_loss: 1.3348\n",
      "Epoch 4/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3629\n",
      "Epoch 4: val_loss did not improve from 1.33482\n",
      "31543/31543 [==============================] - 22s 686us/sample - loss: 1.3629 - val_loss: 1.3460\n",
      "Epoch 5/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3643\n",
      "Epoch 5: val_loss did not improve from 1.33482\n",
      "31543/31543 [==============================] - 19s 617us/sample - loss: 1.3643 - val_loss: 1.3406\n",
      "Epoch 6/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3589\n",
      "Epoch 6: val_loss improved from 1.33482 to 1.33266, saving model to ./checkpoints/unknown_person_baseline_p12_21.h5\n",
      "31543/31543 [==============================] - 22s 682us/sample - loss: 1.3589 - val_loss: 1.3327\n",
      "Epoch 7/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3611\n",
      "Epoch 7: val_loss improved from 1.33266 to 1.33105, saving model to ./checkpoints/unknown_person_baseline_p12_21.h5\n",
      "31543/31543 [==============================] - 20s 648us/sample - loss: 1.3611 - val_loss: 1.3310\n",
      "Epoch 8/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3705\n",
      "Epoch 8: val_loss did not improve from 1.33105\n",
      "31543/31543 [==============================] - 19s 611us/sample - loss: 1.3705 - val_loss: 1.3388\n",
      "Epoch 9/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3685\n",
      "Epoch 9: val_loss did not improve from 1.33105\n",
      "31543/31543 [==============================] - 20s 626us/sample - loss: 1.3685 - val_loss: 1.3412\n",
      "Epoch 10/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3761\n",
      "Epoch 10: val_loss improved from 1.33105 to 1.33089, saving model to ./checkpoints/unknown_person_baseline_p12_21.h5\n",
      "31543/31543 [==============================] - 20s 625us/sample - loss: 1.3761 - val_loss: 1.3309\n",
      "Epoch 11/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3787\n",
      "Epoch 11: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 20s 644us/sample - loss: 1.3787 - val_loss: 1.3381\n",
      "Epoch 12/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3962\n",
      "Epoch 12: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 20s 622us/sample - loss: 1.3962 - val_loss: 1.3435\n",
      "Epoch 13/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3882\n",
      "Epoch 13: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 20s 630us/sample - loss: 1.3882 - val_loss: 1.3391\n",
      "Epoch 14/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3887\n",
      "Epoch 14: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 19s 617us/sample - loss: 1.3887 - val_loss: 1.3419\n",
      "Epoch 15/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3803\n",
      "Epoch 15: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 19s 617us/sample - loss: 1.3803 - val_loss: 1.3345\n",
      "Epoch 16/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4058\n",
      "Epoch 16: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 19s 611us/sample - loss: 1.4058 - val_loss: 1.3529\n",
      "Epoch 17/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3935\n",
      "Epoch 17: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 20s 623us/sample - loss: 1.3935 - val_loss: 1.3389\n",
      "Epoch 18/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4066\n",
      "Epoch 18: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 20s 626us/sample - loss: 1.4066 - val_loss: 1.3541\n",
      "Epoch 19/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3967\n",
      "Epoch 19: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 22s 688us/sample - loss: 1.3967 - val_loss: 1.3570\n",
      "Epoch 20/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4344\n",
      "Epoch 20: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 22s 702us/sample - loss: 1.4344 - val_loss: 1.3559\n",
      "Epoch 21/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4355\n",
      "Epoch 21: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 22s 691us/sample - loss: 1.4355 - val_loss: 1.3538\n",
      "Epoch 22/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4446\n",
      "Epoch 22: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 22s 699us/sample - loss: 1.4446 - val_loss: 1.3538\n",
      "Epoch 23/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4162\n",
      "Epoch 23: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 22s 693us/sample - loss: 1.4162 - val_loss: 1.3570\n",
      "Epoch 24/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4092\n",
      "Epoch 24: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 21s 671us/sample - loss: 1.4092 - val_loss: 1.3568\n",
      "Epoch 25/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4021\n",
      "Epoch 25: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 20s 635us/sample - loss: 1.4021 - val_loss: 1.3541\n",
      "Epoch 26/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3948\n",
      "Epoch 26: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 19s 605us/sample - loss: 1.3948 - val_loss: 1.3477\n",
      "Epoch 27/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3953\n",
      "Epoch 27: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 19s 609us/sample - loss: 1.3953 - val_loss: 1.3527\n",
      "Epoch 28/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3917\n",
      "Epoch 28: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 19s 616us/sample - loss: 1.3917 - val_loss: 1.3462\n",
      "Epoch 29/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3894\n",
      "Epoch 29: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 21s 664us/sample - loss: 1.3894 - val_loss: 1.3521\n",
      "Epoch 30/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3852\n",
      "Epoch 30: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 22s 689us/sample - loss: 1.3852 - val_loss: 1.3437\n",
      "Epoch 31/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3880\n",
      "Epoch 31: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 22s 689us/sample - loss: 1.3880 - val_loss: 1.3464\n",
      "Epoch 32/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3852\n",
      "Epoch 32: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 22s 693us/sample - loss: 1.3852 - val_loss: 1.3482\n",
      "Epoch 33/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3822\n",
      "Epoch 33: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 20s 646us/sample - loss: 1.3822 - val_loss: 1.3434\n",
      "Epoch 34/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3798\n",
      "Epoch 34: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 21s 668us/sample - loss: 1.3798 - val_loss: 1.3443\n",
      "Epoch 35/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3793\n",
      "Epoch 35: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 22s 685us/sample - loss: 1.3793 - val_loss: 1.3488\n",
      "Epoch 36/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3772\n",
      "Epoch 36: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 19s 595us/sample - loss: 1.3772 - val_loss: 1.3406\n",
      "Epoch 37/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3754\n",
      "Epoch 37: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 21s 668us/sample - loss: 1.3754 - val_loss: 1.3403\n",
      "Epoch 38/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3748\n",
      "Epoch 38: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 22s 690us/sample - loss: 1.3748 - val_loss: 1.3456\n",
      "Epoch 39/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3749\n",
      "Epoch 39: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 22s 693us/sample - loss: 1.3749 - val_loss: 1.3444\n",
      "Epoch 40/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3752\n",
      "Epoch 40: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 22s 695us/sample - loss: 1.3752 - val_loss: 1.3368\n",
      "Epoch 41/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3729\n",
      "Epoch 41: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 21s 658us/sample - loss: 1.3729 - val_loss: 1.3430\n",
      "Epoch 42/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3736\n",
      "Epoch 42: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 20s 619us/sample - loss: 1.3736 - val_loss: 1.3393\n",
      "Epoch 43/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3684\n",
      "Epoch 43: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 20s 619us/sample - loss: 1.3684 - val_loss: 1.3347\n",
      "Epoch 44/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3687\n",
      "Epoch 44: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 21s 664us/sample - loss: 1.3687 - val_loss: 1.3517\n",
      "Epoch 45/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3657\n",
      "Epoch 45: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 22s 683us/sample - loss: 1.3657 - val_loss: 1.3383\n",
      "Epoch 46/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3643\n",
      "Epoch 46: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 19s 611us/sample - loss: 1.3643 - val_loss: 1.3397\n",
      "Epoch 47/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3650\n",
      "Epoch 47: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 19s 611us/sample - loss: 1.3650 - val_loss: 1.3379\n",
      "Epoch 48/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3637\n",
      "Epoch 48: val_loss did not improve from 1.33089\n",
      "31543/31543 [==============================] - 19s 615us/sample - loss: 1.3637 - val_loss: 1.3341\n",
      "Epoch 49/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3606\n",
      "Epoch 49: val_loss improved from 1.33089 to 1.32988, saving model to ./checkpoints/unknown_person_baseline_p12_21.h5\n",
      "31543/31543 [==============================] - 20s 619us/sample - loss: 1.3606 - val_loss: 1.3299\n",
      "Epoch 50/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3617\n",
      "Epoch 50: val_loss did not improve from 1.32988\n",
      "31543/31543 [==============================] - 20s 621us/sample - loss: 1.3617 - val_loss: 1.3335\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 17:31:55.126315: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_32_3/lstm_cell_143/recurrent_kernel/Assign' id:62540 op device:{requested: '', assigned: ''} def:{{{node lstm_32_3/lstm_cell_143/recurrent_kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_32_3/lstm_cell_143/recurrent_kernel, lstm_32_3/lstm_cell_143/recurrent_kernel/Initializer/mul_1)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 17:31:58.316903: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_35_3/lstm_cell_146/bias/v/Assign' id:65188 op device:{requested: '', assigned: ''} def:{{{node lstm_35_3/lstm_cell_146/bias/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_35_3/lstm_cell_146/bias/v, lstm_35_3/lstm_cell_146/bias/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 17:32:01.371296: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_3/cond/Merge' id:63349 op device:{requested: '', assigned: ''} def:{{{node dropout_4_3/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_3/cond/Identity, dropout_4_3/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1944)\n",
      "(1514, 1944)\n",
      "(1644, 1944)\n",
      "(1764, 1944)\n",
      "(1836, 1944)\n",
      "(1699, 1944)\n",
      "(1369, 1944)\n",
      "(1766, 1944)\n",
      "(1631, 1944)\n",
      "(1538, 1944)\n",
      "(1932, 1944)\n",
      "(1679, 1944)\n",
      "(1776, 1944)\n",
      "(1872, 1944)\n",
      "(1692, 1944)\n",
      "(1824, 1944)\n",
      "(970, 1944)\n",
      "(1656, 1944)\n",
      "(1896, 1944)\n",
      "{1: 7.0239903040187714, 2: 5.675349359770153, 4: 9.451634367434055, 5: 6.266781751419996, 6: 5.415888945482059, 8: 9.587039112876258, 9: 6.714894144206619, 10: 7.9845862556198695, 11: 6.333396695749278, 13: 7.61648715022117, 17: 10.0, 19: 8.020596982871215, 21: 9.79114635884218, 22: 1.0, 25: 8.095729095210228, 26: 7.299541595327224, 27: 4.46192324389249, 28: 6.395314255743758, 29: 2.678369143308937}\n",
      "Train on 31543 samples, validate on 3514 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 17:35:58.544520: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_7/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31543/31543 [==============================] - ETA: 0s - loss: 9.8825\n",
      "Epoch 1: val_loss improved from inf to 1.36055, saving model to ./checkpoints/unknown_person_p12_21.h5\n",
      "31543/31543 [==============================] - 31s 980us/sample - loss: 9.8825 - val_loss: 1.3606\n",
      "Epoch 2/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.8523\n",
      "Epoch 2: val_loss improved from 1.36055 to 1.34671, saving model to ./checkpoints/unknown_person_p12_21.h5\n",
      "31543/31543 [==============================] - 22s 692us/sample - loss: 9.8523 - val_loss: 1.3467\n",
      "Epoch 3/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.7617\n",
      "Epoch 3: val_loss improved from 1.34671 to 1.34268, saving model to ./checkpoints/unknown_person_p12_21.h5\n",
      "31543/31543 [==============================] - 22s 691us/sample - loss: 9.7617 - val_loss: 1.3427\n",
      "Epoch 4/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.6778\n",
      "Epoch 4: val_loss improved from 1.34268 to 1.33809, saving model to ./checkpoints/unknown_person_p12_21.h5\n",
      "31543/31543 [==============================] - 21s 675us/sample - loss: 9.6778 - val_loss: 1.3381\n",
      "Epoch 5/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.6885\n",
      "Epoch 5: val_loss improved from 1.33809 to 1.33439, saving model to ./checkpoints/unknown_person_p12_21.h5\n",
      "31543/31543 [==============================] - 19s 614us/sample - loss: 9.6885 - val_loss: 1.3344\n",
      "Epoch 6/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.6540\n",
      "Epoch 6: val_loss improved from 1.33439 to 1.33305, saving model to ./checkpoints/unknown_person_p12_21.h5\n",
      "31543/31543 [==============================] - 19s 606us/sample - loss: 9.6540 - val_loss: 1.3331\n",
      "Epoch 7/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.6421\n",
      "Epoch 7: val_loss did not improve from 1.33305\n",
      "31543/31543 [==============================] - 22s 691us/sample - loss: 9.6421 - val_loss: 1.3384\n",
      "Epoch 8/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.6014\n",
      "Epoch 8: val_loss did not improve from 1.33305\n",
      "31543/31543 [==============================] - 22s 688us/sample - loss: 9.6014 - val_loss: 1.3364\n",
      "Epoch 9/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.6130\n",
      "Epoch 9: val_loss did not improve from 1.33305\n",
      "31543/31543 [==============================] - 22s 692us/sample - loss: 9.6130 - val_loss: 1.3349\n",
      "Epoch 10/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.6087\n",
      "Epoch 10: val_loss improved from 1.33305 to 1.33212, saving model to ./checkpoints/unknown_person_p12_21.h5\n",
      "31543/31543 [==============================] - 22s 693us/sample - loss: 9.6087 - val_loss: 1.3321\n",
      "Epoch 11/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.5552\n",
      "Epoch 11: val_loss did not improve from 1.33212\n",
      "31543/31543 [==============================] - 20s 643us/sample - loss: 9.5552 - val_loss: 1.3395\n",
      "Epoch 12/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.5523\n",
      "Epoch 12: val_loss did not improve from 1.33212\n",
      "31543/31543 [==============================] - 20s 625us/sample - loss: 9.5523 - val_loss: 1.3330\n",
      "Epoch 13/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.5473\n",
      "Epoch 13: val_loss did not improve from 1.33212\n",
      "31543/31543 [==============================] - 20s 638us/sample - loss: 9.5473 - val_loss: 1.3363\n",
      "Epoch 14/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.5321\n",
      "Epoch 14: val_loss improved from 1.33212 to 1.32559, saving model to ./checkpoints/unknown_person_p12_21.h5\n",
      "31543/31543 [==============================] - 21s 651us/sample - loss: 9.5321 - val_loss: 1.3256\n",
      "Epoch 15/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.5238\n",
      "Epoch 15: val_loss improved from 1.32559 to 1.32307, saving model to ./checkpoints/unknown_person_p12_21.h5\n",
      "31543/31543 [==============================] - 22s 693us/sample - loss: 9.5238 - val_loss: 1.3231\n",
      "Epoch 16/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.4917\n",
      "Epoch 16: val_loss did not improve from 1.32307\n",
      "31543/31543 [==============================] - 20s 620us/sample - loss: 9.4917 - val_loss: 1.3335\n",
      "Epoch 17/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.4875\n",
      "Epoch 17: val_loss did not improve from 1.32307\n",
      "31543/31543 [==============================] - 21s 676us/sample - loss: 9.4875 - val_loss: 1.3294\n",
      "Epoch 18/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.4663\n",
      "Epoch 18: val_loss did not improve from 1.32307\n",
      "31543/31543 [==============================] - 20s 621us/sample - loss: 9.4663 - val_loss: 1.3520\n",
      "Epoch 19/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.4890\n",
      "Epoch 19: val_loss did not improve from 1.32307\n",
      "31543/31543 [==============================] - 20s 641us/sample - loss: 9.4890 - val_loss: 1.3335\n",
      "Epoch 20/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.4720\n",
      "Epoch 20: val_loss did not improve from 1.32307\n",
      "31543/31543 [==============================] - 22s 685us/sample - loss: 9.4720 - val_loss: 1.3255\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 17:43:07.025324: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_18_4/lstm_cell_166/recurrent_kernel/Assign' id:79696 op device:{requested: '', assigned: ''} def:{{{node lstm_18_4/lstm_cell_166/recurrent_kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_18_4/lstm_cell_166/recurrent_kernel, lstm_18_4/lstm_cell_166/recurrent_kernel/Initializer/mul_1)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 17:43:11.175412: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_21_4/lstm_cell_169/recurrent_kernel/m/Assign' id:83728 op device:{requested: '', assigned: ''} def:{{{node lstm_21_4/lstm_cell_169/recurrent_kernel/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_21_4/lstm_cell_169/recurrent_kernel/m, lstm_21_4/lstm_cell_169/recurrent_kernel/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31543 samples, validate on 3514 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 17:43:17.736241: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_9/mul' id:83337 op device:{requested: '', assigned: ''} def:{{{node loss_9/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_9/mul/x, loss_9/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 17:43:39.331084: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_8/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3625"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 17:44:02.204074: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_9/mul' id:83337 op device:{requested: '', assigned: ''} def:{{{node loss_9/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_9/mul/x, loss_9/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.33866, saving model to ./checkpoints/unknown_person_baseline_2_p12_21.h5\n",
      "31543/31543 [==============================] - 32s 1ms/sample - loss: 1.3625 - val_loss: 1.3387\n",
      "Epoch 2/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3597\n",
      "Epoch 2: val_loss improved from 1.33866 to 1.33261, saving model to ./checkpoints/unknown_person_baseline_2_p12_21.h5\n",
      "31543/31543 [==============================] - 22s 699us/sample - loss: 1.3597 - val_loss: 1.3326\n",
      "Epoch 3/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3569\n",
      "Epoch 3: val_loss improved from 1.33261 to 1.33178, saving model to ./checkpoints/unknown_person_baseline_2_p12_21.h5\n",
      "31543/31543 [==============================] - 22s 692us/sample - loss: 1.3569 - val_loss: 1.3318\n",
      "Epoch 4/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3569\n",
      "Epoch 4: val_loss improved from 1.33178 to 1.33000, saving model to ./checkpoints/unknown_person_baseline_2_p12_21.h5\n",
      "31543/31543 [==============================] - 22s 685us/sample - loss: 1.3569 - val_loss: 1.3300\n",
      "Epoch 5/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3553\n",
      "Epoch 5: val_loss did not improve from 1.33000\n",
      "31543/31543 [==============================] - 22s 686us/sample - loss: 1.3553 - val_loss: 1.3310\n",
      "Epoch 6/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3538\n",
      "Epoch 6: val_loss did not improve from 1.33000\n",
      "31543/31543 [==============================] - 21s 659us/sample - loss: 1.3538 - val_loss: 1.3302\n",
      "Epoch 7/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3503\n",
      "Epoch 7: val_loss did not improve from 1.33000\n",
      "31543/31543 [==============================] - 20s 643us/sample - loss: 1.3503 - val_loss: 1.3331\n",
      "Epoch 8/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3491\n",
      "Epoch 8: val_loss improved from 1.33000 to 1.32974, saving model to ./checkpoints/unknown_person_baseline_2_p12_21.h5\n",
      "31543/31543 [==============================] - 19s 614us/sample - loss: 1.3491 - val_loss: 1.3297\n",
      "Epoch 9/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3508\n",
      "Epoch 9: val_loss improved from 1.32974 to 1.32822, saving model to ./checkpoints/unknown_person_baseline_2_p12_21.h5\n",
      "31543/31543 [==============================] - 19s 611us/sample - loss: 1.3508 - val_loss: 1.3282\n",
      "Epoch 10/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3491\n",
      "Epoch 10: val_loss did not improve from 1.32822\n",
      "31543/31543 [==============================] - 19s 602us/sample - loss: 1.3491 - val_loss: 1.3285\n",
      "Epoch 11/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3488\n",
      "Epoch 11: val_loss did not improve from 1.32822\n",
      "31543/31543 [==============================] - 19s 605us/sample - loss: 1.3488 - val_loss: 1.3305\n",
      "Epoch 12/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3426\n",
      "Epoch 12: val_loss did not improve from 1.32822\n",
      "31543/31543 [==============================] - 19s 613us/sample - loss: 1.3426 - val_loss: 1.3304\n",
      "Epoch 13/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3446\n",
      "Epoch 13: val_loss did not improve from 1.32822\n",
      "31543/31543 [==============================] - 21s 653us/sample - loss: 1.3446 - val_loss: 1.3296\n",
      "Epoch 14/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3437\n",
      "Epoch 14: val_loss improved from 1.32822 to 1.32681, saving model to ./checkpoints/unknown_person_baseline_2_p12_21.h5\n",
      "31543/31543 [==============================] - 22s 697us/sample - loss: 1.3437 - val_loss: 1.3268\n",
      "Epoch 15/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3434\n",
      "Epoch 15: val_loss did not improve from 1.32681\n",
      "31543/31543 [==============================] - 22s 694us/sample - loss: 1.3434 - val_loss: 1.3288\n",
      "Epoch 16/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3414\n",
      "Epoch 16: val_loss did not improve from 1.32681\n",
      "31543/31543 [==============================] - 22s 691us/sample - loss: 1.3414 - val_loss: 1.3286\n",
      "Epoch 17/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3411\n",
      "Epoch 17: val_loss did not improve from 1.32681\n",
      "31543/31543 [==============================] - 20s 627us/sample - loss: 1.3411 - val_loss: 1.3278\n",
      "Epoch 18/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3437\n",
      "Epoch 18: val_loss improved from 1.32681 to 1.32514, saving model to ./checkpoints/unknown_person_baseline_2_p12_21.h5\n",
      "31543/31543 [==============================] - 20s 646us/sample - loss: 1.3437 - val_loss: 1.3251\n",
      "Epoch 19/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3401\n",
      "Epoch 19: val_loss did not improve from 1.32514\n",
      "31543/31543 [==============================] - 19s 593us/sample - loss: 1.3401 - val_loss: 1.3270\n",
      "Epoch 20/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3391\n",
      "Epoch 20: val_loss did not improve from 1.32514\n",
      "31543/31543 [==============================] - 21s 666us/sample - loss: 1.3391 - val_loss: 1.3377\n",
      "Train on 31543 samples, validate on 3514 samples\n",
      "Epoch 1/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3362\n",
      "Epoch 1: val_loss improved from inf to 1.32258, saving model to ./checkpoints/unknown_person_baseline_p12_22.h5\n",
      "31543/31543 [==============================] - 19s 600us/sample - loss: 1.3362 - val_loss: 1.3226\n",
      "Epoch 2/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3329\n",
      "Epoch 2: val_loss did not improve from 1.32258\n",
      "31543/31543 [==============================] - 18s 586us/sample - loss: 1.3329 - val_loss: 1.3334\n",
      "Epoch 3/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3333\n",
      "Epoch 3: val_loss did not improve from 1.32258\n",
      "31543/31543 [==============================] - 19s 602us/sample - loss: 1.3333 - val_loss: 1.3246\n",
      "Epoch 4/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3334\n",
      "Epoch 4: val_loss did not improve from 1.32258\n",
      "31543/31543 [==============================] - 22s 682us/sample - loss: 1.3334 - val_loss: 1.3269\n",
      "Epoch 5/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3332\n",
      "Epoch 5: val_loss did not improve from 1.32258\n",
      "31543/31543 [==============================] - 22s 682us/sample - loss: 1.3332 - val_loss: 1.3275\n",
      "Epoch 6/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3305\n",
      "Epoch 6: val_loss did not improve from 1.32258\n",
      "31543/31543 [==============================] - 21s 681us/sample - loss: 1.3305 - val_loss: 1.3301\n",
      "Epoch 7/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3292\n",
      "Epoch 7: val_loss did not improve from 1.32258\n",
      "31543/31543 [==============================] - 21s 682us/sample - loss: 1.3292 - val_loss: 1.3264\n",
      "Epoch 8/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3334\n",
      "Epoch 8: val_loss improved from 1.32258 to 1.32036, saving model to ./checkpoints/unknown_person_baseline_p12_22.h5\n",
      "31543/31543 [==============================] - 21s 678us/sample - loss: 1.3334 - val_loss: 1.3204\n",
      "Epoch 9/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3384\n",
      "Epoch 9: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 20s 622us/sample - loss: 1.3384 - val_loss: 1.3273\n",
      "Epoch 10/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3398\n",
      "Epoch 10: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 19s 595us/sample - loss: 1.3398 - val_loss: 1.3208\n",
      "Epoch 11/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3744\n",
      "Epoch 11: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 21s 668us/sample - loss: 1.3744 - val_loss: 1.3325\n",
      "Epoch 12/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3584\n",
      "Epoch 12: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 20s 639us/sample - loss: 1.3584 - val_loss: 1.3309\n",
      "Epoch 13/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3526\n",
      "Epoch 13: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 21s 681us/sample - loss: 1.3526 - val_loss: 1.3270\n",
      "Epoch 14/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3519\n",
      "Epoch 14: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 21s 655us/sample - loss: 1.3519 - val_loss: 1.3340\n",
      "Epoch 15/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3621\n",
      "Epoch 15: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 20s 648us/sample - loss: 1.3621 - val_loss: 1.3412\n",
      "Epoch 16/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3563\n",
      "Epoch 16: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 22s 687us/sample - loss: 1.3563 - val_loss: 1.3421\n",
      "Epoch 17/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3647\n",
      "Epoch 17: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 22s 689us/sample - loss: 1.3647 - val_loss: 1.3430\n",
      "Epoch 18/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3675\n",
      "Epoch 18: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 22s 682us/sample - loss: 1.3675 - val_loss: 1.3416\n",
      "Epoch 19/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3745\n",
      "Epoch 19: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 22s 684us/sample - loss: 1.3745 - val_loss: 1.3416\n",
      "Epoch 20/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3704\n",
      "Epoch 20: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 21s 673us/sample - loss: 1.3704 - val_loss: 1.3449\n",
      "Epoch 21/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3778\n",
      "Epoch 21: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 22s 683us/sample - loss: 1.3778 - val_loss: 1.3399\n",
      "Epoch 22/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3856\n",
      "Epoch 22: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 21s 662us/sample - loss: 1.3856 - val_loss: 1.3437\n",
      "Epoch 23/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3889\n",
      "Epoch 23: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 22s 685us/sample - loss: 1.3889 - val_loss: 1.3396\n",
      "Epoch 24/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3933\n",
      "Epoch 24: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 19s 616us/sample - loss: 1.3933 - val_loss: 1.3552\n",
      "Epoch 25/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3908\n",
      "Epoch 25: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 19s 600us/sample - loss: 1.3908 - val_loss: 1.3455\n",
      "Epoch 26/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3872\n",
      "Epoch 26: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 19s 601us/sample - loss: 1.3872 - val_loss: 1.3449\n",
      "Epoch 27/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3858\n",
      "Epoch 27: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 19s 599us/sample - loss: 1.3858 - val_loss: 1.3420\n",
      "Epoch 28/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3868\n",
      "Epoch 28: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 19s 596us/sample - loss: 1.3868 - val_loss: 1.3413\n",
      "Epoch 29/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3919\n",
      "Epoch 29: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 19s 597us/sample - loss: 1.3919 - val_loss: 1.3443\n",
      "Epoch 30/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.4001\n",
      "Epoch 30: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 19s 602us/sample - loss: 1.4001 - val_loss: 1.3484\n",
      "Epoch 31/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3962\n",
      "Epoch 31: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 19s 613us/sample - loss: 1.3962 - val_loss: 1.3527\n",
      "Epoch 32/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3934\n",
      "Epoch 32: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 20s 620us/sample - loss: 1.3934 - val_loss: 1.3515\n",
      "Epoch 33/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3874\n",
      "Epoch 33: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 19s 603us/sample - loss: 1.3874 - val_loss: 1.3544\n",
      "Epoch 34/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3854\n",
      "Epoch 34: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 19s 599us/sample - loss: 1.3854 - val_loss: 1.3437\n",
      "Epoch 35/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3853\n",
      "Epoch 35: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 19s 598us/sample - loss: 1.3853 - val_loss: 1.3532\n",
      "Epoch 36/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3858\n",
      "Epoch 36: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 19s 604us/sample - loss: 1.3858 - val_loss: 1.3501\n",
      "Epoch 37/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3805\n",
      "Epoch 37: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 20s 645us/sample - loss: 1.3805 - val_loss: 1.3480\n",
      "Epoch 38/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3802\n",
      "Epoch 38: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 21s 681us/sample - loss: 1.3802 - val_loss: 1.3433\n",
      "Epoch 39/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3815\n",
      "Epoch 39: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 22s 685us/sample - loss: 1.3815 - val_loss: 1.3450\n",
      "Epoch 40/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3803\n",
      "Epoch 40: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 20s 627us/sample - loss: 1.3803 - val_loss: 1.3422\n",
      "Epoch 41/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3760\n",
      "Epoch 41: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 18s 586us/sample - loss: 1.3760 - val_loss: 1.3414\n",
      "Epoch 42/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3774\n",
      "Epoch 42: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 21s 670us/sample - loss: 1.3774 - val_loss: 1.3385\n",
      "Epoch 43/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3739\n",
      "Epoch 43: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 22s 684us/sample - loss: 1.3739 - val_loss: 1.3430\n",
      "Epoch 44/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3718\n",
      "Epoch 44: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 19s 613us/sample - loss: 1.3718 - val_loss: 1.3451\n",
      "Epoch 45/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3751\n",
      "Epoch 45: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 20s 619us/sample - loss: 1.3751 - val_loss: 1.3452\n",
      "Epoch 46/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3734\n",
      "Epoch 46: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 21s 676us/sample - loss: 1.3734 - val_loss: 1.3428\n",
      "Epoch 47/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3722\n",
      "Epoch 47: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 19s 610us/sample - loss: 1.3722 - val_loss: 1.3347\n",
      "Epoch 48/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3715\n",
      "Epoch 48: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 19s 599us/sample - loss: 1.3715 - val_loss: 1.3438\n",
      "Epoch 49/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3713\n",
      "Epoch 49: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 19s 594us/sample - loss: 1.3713 - val_loss: 1.3435\n",
      "Epoch 50/50\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3722\n",
      "Epoch 50: val_loss did not improve from 1.32036\n",
      "31543/31543 [==============================] - 19s 593us/sample - loss: 1.3722 - val_loss: 1.3412\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:07:30.583405: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_15_5/lstm_cell_200/kernel/Assign' id:98556 op device:{requested: '', assigned: ''} def:{{{node lstm_15_5/lstm_cell_200/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_15_5/lstm_cell_200/kernel, lstm_15_5/lstm_cell_200/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 18:07:35.577535: W tensorflow/c/c_api.cc:304] Operation '{name:'dense_8/bias/v/Assign' id:103973 op device:{requested: '', assigned: ''} def:{{{node dense_8/bias/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](dense_8/bias/v, dense_8/bias/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 18:07:40.426267: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_5/cond/Merge' id:102107 op device:{requested: '', assigned: ''} def:{{{node dropout_4_5/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_5/cond/Identity, dropout_4_5/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1944)\n",
      "(1514, 1944)\n",
      "(1644, 1944)\n",
      "(1764, 1944)\n",
      "(1836, 1944)\n",
      "(1699, 1944)\n",
      "(1369, 1944)\n",
      "(1766, 1944)\n",
      "(1631, 1944)\n",
      "(1538, 1944)\n",
      "(1932, 1944)\n",
      "(1679, 1944)\n",
      "(1776, 1944)\n",
      "(1872, 1944)\n",
      "(1692, 1944)\n",
      "(1824, 1944)\n",
      "(970, 1944)\n",
      "(1656, 1944)\n",
      "(1896, 1944)\n",
      "{1: 6.935595825492218, 2: 5.141740798942194, 4: 9.535434570488656, 5: 6.411300770785493, 6: 5.249005244563097, 8: 9.559798506755175, 9: 6.70795695280061, 10: 7.817634350503943, 11: 6.304235670580023, 13: 7.716457491016452, 17: 10.0, 19: 7.995716254957417, 21: 9.704955193426299, 22: 1.0, 25: 8.01805200264116, 26: 6.764155747834762, 27: 4.3601950161968155, 28: 6.34772468518778, 29: 2.474511221989075}\n",
      "Train on 31543 samples, validate on 3514 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:11:35.552199: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_9/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31543/31543 [==============================] - ETA: 0s - loss: 9.6325\n",
      "Epoch 1: val_loss improved from inf to 1.35817, saving model to ./checkpoints/unknown_person_p12_22.h5\n",
      "31543/31543 [==============================] - 32s 1000us/sample - loss: 9.6325 - val_loss: 1.3582\n",
      "Epoch 2/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.5797\n",
      "Epoch 2: val_loss improved from 1.35817 to 1.33901, saving model to ./checkpoints/unknown_person_p12_22.h5\n",
      "31543/31543 [==============================] - 22s 684us/sample - loss: 9.5797 - val_loss: 1.3390\n",
      "Epoch 3/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.6156\n",
      "Epoch 3: val_loss did not improve from 1.33901\n",
      "31543/31543 [==============================] - 22s 693us/sample - loss: 9.6156 - val_loss: 1.3494\n",
      "Epoch 4/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.5032\n",
      "Epoch 4: val_loss improved from 1.33901 to 1.33097, saving model to ./checkpoints/unknown_person_p12_22.h5\n",
      "31543/31543 [==============================] - 22s 691us/sample - loss: 9.5032 - val_loss: 1.3310\n",
      "Epoch 5/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.4454\n",
      "Epoch 5: val_loss did not improve from 1.33097\n",
      "31543/31543 [==============================] - 19s 608us/sample - loss: 9.4454 - val_loss: 1.3565\n",
      "Epoch 6/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.4311\n",
      "Epoch 6: val_loss did not improve from 1.33097\n",
      "31543/31543 [==============================] - 19s 604us/sample - loss: 9.4311 - val_loss: 1.3348\n",
      "Epoch 7/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.4206\n",
      "Epoch 7: val_loss improved from 1.33097 to 1.32875, saving model to ./checkpoints/unknown_person_p12_22.h5\n",
      "31543/31543 [==============================] - 19s 614us/sample - loss: 9.4206 - val_loss: 1.3288\n",
      "Epoch 8/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.3882\n",
      "Epoch 8: val_loss improved from 1.32875 to 1.32081, saving model to ./checkpoints/unknown_person_p12_22.h5\n",
      "31543/31543 [==============================] - 19s 607us/sample - loss: 9.3882 - val_loss: 1.3208\n",
      "Epoch 9/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.3665\n",
      "Epoch 9: val_loss did not improve from 1.32081\n",
      "31543/31543 [==============================] - 19s 604us/sample - loss: 9.3665 - val_loss: 1.3270\n",
      "Epoch 10/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.3617\n",
      "Epoch 10: val_loss did not improve from 1.32081\n",
      "31543/31543 [==============================] - 19s 602us/sample - loss: 9.3617 - val_loss: 1.3326\n",
      "Epoch 11/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.3619\n",
      "Epoch 11: val_loss did not improve from 1.32081\n",
      "31543/31543 [==============================] - 19s 605us/sample - loss: 9.3619 - val_loss: 1.3447\n",
      "Epoch 12/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.3505\n",
      "Epoch 12: val_loss did not improve from 1.32081\n",
      "31543/31543 [==============================] - 19s 601us/sample - loss: 9.3505 - val_loss: 1.3275\n",
      "Epoch 13/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.3282\n",
      "Epoch 13: val_loss did not improve from 1.32081\n",
      "31543/31543 [==============================] - 22s 684us/sample - loss: 9.3282 - val_loss: 1.3305\n",
      "Epoch 14/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.3452\n",
      "Epoch 14: val_loss did not improve from 1.32081\n",
      "31543/31543 [==============================] - 22s 687us/sample - loss: 9.3452 - val_loss: 1.3287\n",
      "Epoch 15/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.3114\n",
      "Epoch 15: val_loss did not improve from 1.32081\n",
      "31543/31543 [==============================] - 22s 690us/sample - loss: 9.3114 - val_loss: 1.3340\n",
      "Epoch 16/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.3077\n",
      "Epoch 16: val_loss did not improve from 1.32081\n",
      "31543/31543 [==============================] - 19s 590us/sample - loss: 9.3077 - val_loss: 1.3422\n",
      "Epoch 17/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.2877\n",
      "Epoch 17: val_loss did not improve from 1.32081\n",
      "31543/31543 [==============================] - 19s 593us/sample - loss: 9.2877 - val_loss: 1.3238\n",
      "Epoch 18/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.3093\n",
      "Epoch 18: val_loss did not improve from 1.32081\n",
      "31543/31543 [==============================] - 21s 671us/sample - loss: 9.3093 - val_loss: 1.3249\n",
      "Epoch 19/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.2832\n",
      "Epoch 19: val_loss did not improve from 1.32081\n",
      "31543/31543 [==============================] - 19s 615us/sample - loss: 9.2832 - val_loss: 1.3250\n",
      "Epoch 20/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 9.2670\n",
      "Epoch 20: val_loss did not improve from 1.32081\n",
      "31543/31543 [==============================] - 19s 606us/sample - loss: 9.2670 - val_loss: 1.3286\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:18:30.123323: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_2_6/lstm_cell_224/recurrent_kernel/Assign' id:115894 op device:{requested: '', assigned: ''} def:{{{node lstm_2_6/lstm_cell_224/recurrent_kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_2_6/lstm_cell_224/recurrent_kernel, lstm_2_6/lstm_cell_224/recurrent_kernel/Initializer/mul_1)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 18:18:36.316554: W tensorflow/c/c_api.cc:304] Operation '{name:'conv2d_2_6/kernel/v/Assign' id:122789 op device:{requested: '', assigned: ''} def:{{{node conv2d_2_6/kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](conv2d_2_6/kernel/v, conv2d_2_6/kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31543 samples, validate on 3514 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:18:45.381121: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_13/mul' id:122095 op device:{requested: '', assigned: ''} def:{{{node loss_13/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_13/mul/x, loss_13/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:19:17.429238: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_10/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3404"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:19:38.622092: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_13/mul' id:122095 op device:{requested: '', assigned: ''} def:{{{node loss_13/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_13/mul/x, loss_13/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.32211, saving model to ./checkpoints/unknown_person_baseline_2_p12_22.h5\n",
      "31543/31543 [==============================] - 34s 1ms/sample - loss: 1.3404 - val_loss: 1.3221\n",
      "Epoch 2/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3338\n",
      "Epoch 2: val_loss improved from 1.32211 to 1.32097, saving model to ./checkpoints/unknown_person_baseline_2_p12_22.h5\n",
      "31543/31543 [==============================] - 22s 696us/sample - loss: 1.3338 - val_loss: 1.3210\n",
      "Epoch 3/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3440\n",
      "Epoch 3: val_loss did not improve from 1.32097\n",
      "31543/31543 [==============================] - 22s 682us/sample - loss: 1.3440 - val_loss: 1.3270\n",
      "Epoch 4/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3357\n",
      "Epoch 4: val_loss did not improve from 1.32097\n",
      "31543/31543 [==============================] - 22s 695us/sample - loss: 1.3357 - val_loss: 1.3241\n",
      "Epoch 5/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3332\n",
      "Epoch 5: val_loss did not improve from 1.32097\n",
      "31543/31543 [==============================] - 22s 702us/sample - loss: 1.3332 - val_loss: 1.3246\n",
      "Epoch 6/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3308\n",
      "Epoch 6: val_loss did not improve from 1.32097\n",
      "31543/31543 [==============================] - 20s 633us/sample - loss: 1.3308 - val_loss: 1.3213\n",
      "Epoch 7/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3331\n",
      "Epoch 7: val_loss improved from 1.32097 to 1.31961, saving model to ./checkpoints/unknown_person_baseline_2_p12_22.h5\n",
      "31543/31543 [==============================] - 19s 594us/sample - loss: 1.3331 - val_loss: 1.3196\n",
      "Epoch 8/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3307\n",
      "Epoch 8: val_loss did not improve from 1.31961\n",
      "31543/31543 [==============================] - 19s 602us/sample - loss: 1.3307 - val_loss: 1.3230\n",
      "Epoch 9/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3281\n",
      "Epoch 9: val_loss did not improve from 1.31961\n",
      "31543/31543 [==============================] - 19s 604us/sample - loss: 1.3281 - val_loss: 1.3207\n",
      "Epoch 10/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3279\n",
      "Epoch 10: val_loss did not improve from 1.31961\n",
      "31543/31543 [==============================] - 21s 672us/sample - loss: 1.3279 - val_loss: 1.3222\n",
      "Epoch 11/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3237\n",
      "Epoch 11: val_loss did not improve from 1.31961\n",
      "31543/31543 [==============================] - 22s 682us/sample - loss: 1.3237 - val_loss: 1.3238\n",
      "Epoch 12/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3260\n",
      "Epoch 12: val_loss improved from 1.31961 to 1.31730, saving model to ./checkpoints/unknown_person_baseline_2_p12_22.h5\n",
      "31543/31543 [==============================] - 21s 655us/sample - loss: 1.3260 - val_loss: 1.3173\n",
      "Epoch 13/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3221\n",
      "Epoch 13: val_loss did not improve from 1.31730\n",
      "31543/31543 [==============================] - 22s 693us/sample - loss: 1.3221 - val_loss: 1.3217\n",
      "Epoch 14/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3224\n",
      "Epoch 14: val_loss did not improve from 1.31730\n",
      "31543/31543 [==============================] - 20s 645us/sample - loss: 1.3224 - val_loss: 1.3219\n",
      "Epoch 15/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3234\n",
      "Epoch 15: val_loss did not improve from 1.31730\n",
      "31543/31543 [==============================] - 19s 604us/sample - loss: 1.3234 - val_loss: 1.3285\n",
      "Epoch 16/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3247\n",
      "Epoch 16: val_loss did not improve from 1.31730\n",
      "31543/31543 [==============================] - 19s 610us/sample - loss: 1.3247 - val_loss: 1.3313\n",
      "Epoch 17/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3211\n",
      "Epoch 17: val_loss did not improve from 1.31730\n",
      "31543/31543 [==============================] - 20s 644us/sample - loss: 1.3211 - val_loss: 1.3307\n",
      "Epoch 18/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3209\n",
      "Epoch 18: val_loss did not improve from 1.31730\n",
      "31543/31543 [==============================] - 19s 596us/sample - loss: 1.3209 - val_loss: 1.3202\n",
      "Epoch 19/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3221\n",
      "Epoch 19: val_loss did not improve from 1.31730\n",
      "31543/31543 [==============================] - 21s 681us/sample - loss: 1.3221 - val_loss: 1.3222\n",
      "Epoch 20/20\n",
      "31543/31543 [==============================] - ETA: 0s - loss: 1.3218\n",
      "Epoch 20: val_loss improved from 1.31730 to 1.31170, saving model to ./checkpoints/unknown_person_baseline_2_p12_22.h5\n",
      "31543/31543 [==============================] - 22s 699us/sample - loss: 1.3218 - val_loss: 1.3117\n"
     ]
    }
   ],
   "source": [
    "person_nums = [1,2,4,5,6,8,9,10,11,12,13,17,19,21,22,25,26,27,28,29]\n",
    "## Build the baseline model for emotion recognition with dropout layers\n",
    "from tensorflow.keras.models import Model\n",
    "from tensorflow.keras.layers import Input, Dense, concatenate, SimpleRNN, LSTM, Conv2D, Flatten, MaxPooling2D, GRU, AveragePooling2D, Dropout, Lambda\n",
    "from tensorflow.keras.callbacks import ModelCheckpoint\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "def individual_model(features_list):\n",
    "    input_layers = []\n",
    "    hidden_layers = []\n",
    "    combined_feature = np.empty((len(features_list[0]),0))\n",
    "    for i, feature in enumerate(features_list):\n",
    "        \n",
    "        if len(feature.shape) == 3:\n",
    "            input_i = Input(shape=(feature.shape[1], feature.shape[2]))\n",
    "            input_layers.append(input_i)\n",
    "\n",
    "            hidden_i = input_i[:,:,:,None]\n",
    "            hidden_i = Conv2D(32, 3, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((3, 3))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(16, 3, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((3, 3))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(8, 1, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((2, 2))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(4, 1, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((2, 2))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Flatten()(hidden_i)\n",
    "\n",
    "            hidden_layers.append(hidden_i)\n",
    "        elif feature.shape[1] <20:\n",
    "            combined_feature = np.hstack((combined_feature, feature))\n",
    "            \n",
    "        else:  # For series features\n",
    "            input_i = Input(shape=(feature.shape[1],))\n",
    "            input_layers.append(input_i)\n",
    "            hidden_i = Lambda(lambda x: x[:, :, None])(input_i)  # Add a new dimension\n",
    "            hidden_i = LSTM(4)(hidden_i)\n",
    "            hidden_layers.append(hidden_i)\n",
    "    input_i = Input(shape=(combined_feature.shape[1],))\n",
    "    input_layers.append(input_i)\n",
    "    dense_num = np.max((1, int(combined_feature.shape[1]/2)))\n",
    "    hidden_i = Dense(dense_num, activation='relu')(input_i)\n",
    "    hidden_layers.append(hidden_i)\n",
    "    print(combined_feature.shape)\n",
    "    concat_layer = concatenate(hidden_layers)\n",
    "    h = Dropout(0.2)(concat_layer)\n",
    "    h = Dense(64, activation='relu')(h)\n",
    "    h = Dense(32, activation='relu')(h)\n",
    "    output_layer = Dense(2)(h)\n",
    "    model = Model(inputs=input_layers, outputs=output_layer)\n",
    "    model.compile(loss='mean_absolute_error', optimizer='adam')\n",
    "    return model\n",
    "\n",
    "import tensorflow as tf\n",
    "from tensorflow.keras.callbacks import Callback\n",
    "import numpy as np\n",
    "\n",
    "class PruningCallback(tf.keras.callbacks.Callback):\n",
    "    def __init__(self, initial_pruning_factor=0.1, final_pruning_factor=0.5, start_epoch=0, end_epoch=None, frequency=1):\n",
    "        super(PruningCallback, self).__init__()\n",
    "        self.initial_pruning_factor = initial_pruning_factor\n",
    "        self.final_pruning_factor = final_pruning_factor\n",
    "        self.start_epoch = start_epoch\n",
    "        self.end_epoch = end_epoch if end_epoch is not None else np.inf\n",
    "        self.frequency = frequency\n",
    "        self.pruned_weights = {}\n",
    "        self.layer_importance = {}\n",
    "\n",
    "    def get_pruning_factor(self, epoch):\n",
    "        if epoch < self.start_epoch:\n",
    "            return 0\n",
    "        if epoch > self.end_epoch:\n",
    "            return self.final_pruning_factor\n",
    "        return self.initial_pruning_factor + (self.final_pruning_factor - self.initial_pruning_factor) * (epoch - self.start_epoch) / (self.end_epoch - self.start_epoch)\n",
    "\n",
    "    def on_train_begin(self, logs=None):\n",
    "        total_weight_magnitude = 0\n",
    "        for layer in self.model.layers:\n",
    "            if hasattr(layer, 'get_weights'):\n",
    "                weights = layer.get_weights()\n",
    "                layer_norm = sum(np.linalg.norm(w) for w in weights)\n",
    "                total_weight_magnitude += layer_norm\n",
    "                self.layer_importance[layer.name] = layer_norm\n",
    "    \n",
    "        # Normalize the layer importance values so they sum up to 1\n",
    "        for layer_name in self.layer_importance:\n",
    "            self.layer_importance[layer_name] /= total_weight_magnitude\n",
    "    # def on_train_begin(self, logs=None):\n",
    "    #     total_weight_magnitude = sum([np.linalg.norm(layer.get_weights()) for layer in self.model.layers if hasattr(layer, 'get_weights')])\n",
    "    #     for layer in self.model.layers:\n",
    "    #         if hasattr(layer, 'get_weights'):\n",
    "    #             self.layer_importance[layer.name] = np.linalg.norm(layer.get_weights()) / total_weight_magnitude\n",
    "\n",
    "    def get_layer_pruning_factor(self, layer_name, global_pruning_factor):\n",
    "        if layer_name in self.layer_importance:\n",
    "            importance = self.layer_importance[layer_name]\n",
    "            adjusted_pruning_factor = global_pruning_factor * (1 - importance)\n",
    "            return min(adjusted_pruning_factor, 1)  # Ensure the pruning factor is not greater than 1\n",
    "        return global_pruning_factor\n",
    "    def prune_weights(self, layer, global_pruning_factor):\n",
    "        \n",
    "        weights = layer.get_weights()\n",
    "        layer_name = layer.name\n",
    "        pruning_factor = self.get_layer_pruning_factor(layer_name, global_pruning_factor)\n",
    "\n",
    "        if layer_name not in self.pruned_weights:\n",
    "            self.pruned_weights[layer_name] = [np.zeros_like(w, dtype=bool) for w in weights]\n",
    "        \n",
    "        for i in range(len(weights)):\n",
    "            weight = weights[i]\n",
    "            # print(weight.shape)\n",
    "            # print(weight.size)\n",
    "            if weight.ndim > 1:  # Only prune dense or convolutional layers\n",
    "                unpruned_weights = np.logical_not(self.pruned_weights[layer_name][i])\n",
    "                num_unpruned = np.sum(unpruned_weights)\n",
    "                num_pruning = min(num_unpruned, int(weight.size * pruning_factor) - np.sum(self.pruned_weights[layer_name][i]))\n",
    "                num_pruning = int(weight.size * pruning_factor) - np.sum(self.pruned_weights[layer_name][i])\n",
    "                if num_pruning > 0:\n",
    "                    unpruned_flat_indices = np.flatnonzero(unpruned_weights)\n",
    "                    abs_unpruned_weights = np.abs(weight[unpruned_weights])\n",
    "                    pruning_flat_indices = np.argpartition(abs_unpruned_weights, num_pruning)[:num_pruning]\n",
    "                    \n",
    "                    indices = np.unravel_index(pruning_flat_indices, weight.shape)\n",
    "                    self.pruned_weights[layer_name][i][indices] = True\n",
    "\n",
    "                weights[i] = weights[i]*(~self.pruned_weights[layer_name][i])\n",
    "                \n",
    "        layer.set_weights(weights)\n",
    "\n",
    "    def on_epoch_end(self, epoch, logs=None):\n",
    "        if (epoch - self.start_epoch) % self.frequency != 0:\n",
    "            return\n",
    "\n",
    "        pruning_factor = self.get_pruning_factor(epoch)\n",
    "        for layer in self.model.layers:\n",
    "            if isinstance(layer, (tf.keras.layers.Dense, tf.keras.layers.Conv2D)):\n",
    "                self.prune_weights(layer, pruning_factor)\n",
    "from feature_emotion import feature_extract, split_data, label_unique_tuples, split_data_unknown\n",
    "gts, sensor_nums, walk_nums, trace_nums, people_nums, spe_centr, delta_spe_centr, spe_crest, delta_spe_crest, spe_decrease, delta_spe_decrease, spe_entropy, delta_spe_entropy, spe_flatness, delta_spe_flatness, spe_flux, delta_spe_flux, spe_kurtosis, delta_spe_kurtosis, spe_skewness, delta_spe_skewness, spe_rfp, delta_spe_rfp, spe_slope, delta_spe_slope, spe_spread, delta_spe_spread, wlk_fre, wlk_fres_trace, cwt_figs_all, cwt_sum_all_0, cwt_sum_all_1, cwt_sum_all_2, cwt_sum_all_3, high_fre_compos, pitchs, low_fre_compos, auto_corrs, real_hils, imag_hils, dur_time_1_alls, dur_time_2_alls, jitters, shimmers, jitter_rap, hrs, feature, slope, zcrs, fft_features, energy_alls, log_energy_alls, smoothe_energy_alls, legendres, double_support_time, pdps_new, lpcs, ceps_features = feature_extract(person_nums)\n",
    "\n",
    "walk_nums_all = np.squeeze(walk_nums)\n",
    "trace_nums_all = np.squeeze(trace_nums)\n",
    "people_nums_all = np.squeeze(people_nums)\n",
    "\n",
    "## 0: train, 1: validation 2: test\n",
    "\n",
    "test_person_id = [12]\n",
    "flag_tr_val_te = split_data_unknown(test_person_id, walk_nums_all, trace_nums_all, people_nums_all, person_nums, rand_seed = 42)\n",
    "## Data Normalization before training ans testing\n",
    "import tensorflow as tf\n",
    "tf.compat.v1.disable_v2_behavior()\n",
    "from tensorflow.keras.models import Model\n",
    "from tensorflow.keras.layers import Input, Dense, concatenate, LSTM\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "scalers = []\n",
    "X_train_normalized = []\n",
    "X_val_normalized = []\n",
    "X_test_normalized = []\n",
    "train_idx = np.where(flag_tr_val_te ==0)[0]\n",
    "np.random.shuffle(train_idx)\n",
    "val_idx = np.where(flag_tr_val_te ==1)[0]\n",
    "test_idx = np.where(flag_tr_val_te ==2)[0]\n",
    "\n",
    "for i, feature in enumerate([spe_centr, delta_spe_centr, spe_crest, delta_spe_crest, spe_decrease, delta_spe_decrease, spe_entropy, delta_spe_entropy, spe_flatness, delta_spe_flatness, spe_flux, delta_spe_flux, spe_kurtosis, delta_spe_kurtosis, spe_skewness, delta_spe_skewness, spe_rfp, delta_spe_rfp, spe_slope, delta_spe_slope, spe_spread, delta_spe_spread, wlk_fre, wlk_fres_trace, cwt_figs_all, cwt_sum_all_0, cwt_sum_all_1, cwt_sum_all_2, cwt_sum_all_3, high_fre_compos, pitchs, low_fre_compos, auto_corrs, real_hils, imag_hils, dur_time_1_alls, dur_time_2_alls, jitters, shimmers, jitter_rap, hrs, feature, slope, zcrs, fft_features, energy_alls, log_energy_alls, smoothe_energy_alls, legendres, double_support_time, pdps_new, lpcs, ceps_features]):\n",
    "    scaler = StandardScaler()\n",
    "    if len(feature.shape)==2:\n",
    "        X_train_i = feature[train_idx,:]\n",
    "        X_val_i = feature[val_idx,:]\n",
    "        X_test_i = feature[test_idx,:]\n",
    "        X_train_normalized_i = scaler.fit_transform(X_train_i)\n",
    "        X_val_normalized_i = scaler.transform(X_val_i)\n",
    "        X_test_normalized_i = scaler.transform(X_test_i)\n",
    "        scalers.append(scaler)\n",
    "    else:\n",
    "        X_train_i = feature[train_idx,:,:]\n",
    "        X_val_i = feature[val_idx,:,:]\n",
    "        X_test_i = feature[test_idx,:,:]\n",
    "        X_train_normalized_i = scaler.fit_transform(X_train_i.reshape(X_train_i.shape[0], -1)).reshape(X_train_i.shape)\n",
    "        X_val_normalized_i = scaler.transform(X_val_i.reshape(X_val_i.shape[0], -1)).reshape(X_val_i.shape)\n",
    "        X_test_normalized_i = scaler.transform(X_test_i.reshape(X_test_i.shape[0], -1)).reshape(X_test_i.shape)\n",
    "        scalers.append(scaler)\n",
    "    X_train_normalized.append(X_train_normalized_i)\n",
    "    X_val_normalized.append(X_val_normalized_i)\n",
    "    X_test_normalized.append(X_test_normalized_i)\n",
    "y_train = gts[train_idx,:]\n",
    "y_val = gts[val_idx,:]\n",
    "y_test = gts[test_idx,:]\n",
    "X_train_normalized_new = []\n",
    "combined_feature = np.empty((len(X_train_normalized[0]),0))\n",
    "for feature in X_train_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_train_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_train_normalized_new.append(feature)\n",
    "X_train_normalized_new.append(combined_feature)\n",
    "\n",
    "X_val_normalized_new = []\n",
    "combined_feature = np.empty((len(X_val_normalized[0]),0))\n",
    "for feature in X_val_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_val_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_val_normalized_new.append(feature)\n",
    "X_val_normalized_new.append(combined_feature)\n",
    "\n",
    "X_test_normalized_new = []\n",
    "combined_feature = np.empty((len(X_test_normalized[0]),0))\n",
    "for feature in X_test_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_test_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_test_normalized_new.append(feature)\n",
    "X_test_normalized_new.append(combined_feature)\n",
    "\n",
    "model = individual_model(X_train_normalized)\n",
    "model.compile(loss='mean_absolute_error', optimizer='adam')\n",
    "num_epochs = 50\n",
    "\n",
    "\n",
    "# Gradually prune weights in dense and convolutional layers from 10% to 50% over the course of training, starting from epoch 0.\n",
    "\n",
    "\n",
    "rates = [0.4, 0.5, 0.6]\n",
    "idx = 20\n",
    "for r in rates:\n",
    "    model_name = './checkpoints/unknown_person_baseline_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=0.01, final_pruning_factor=r, start_epoch=5, end_epoch=20, frequency=1)\n",
    "    model.fit(X_train_normalized_new, y_train, epochs=num_epochs, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])\n",
    "    import tensorflow as tf\n",
    "    model = tf.keras.models.load_model(model_name)\n",
    "    from tensorflow.keras.models import Model\n",
    "    layers = model.layers\n",
    "    second_last_layer_output = layers[-4].output\n",
    "    feature_extractor_model = Model(inputs=model.input, outputs=second_last_layer_output)\n",
    "    train_features = feature_extractor_model.predict(X_train_normalized_new)\n",
    "    test_features = feature_extractor_model.predict(X_test_normalized_new)\n",
    "    \n",
    "    p_train = people_nums[train_idx,:]\n",
    "    p_val = people_nums[val_idx,:]\n",
    "    p_test = people_nums[test_idx,:]\n",
    "    ## Calculate the distance between test person and training person\n",
    "    def euclidean_distance(a, b):\n",
    "        return np.sqrt(np.sum((a - b) ** 2, axis=1))\n",
    "    \n",
    "    distance_dict = {}\n",
    "    for ii in range(len(person_nums)):\n",
    "        if person_nums[ii] == test_person_id[0]:\n",
    "            continue\n",
    "        else:\n",
    "            \n",
    "            ind = np.where(p_train ==person_nums[ii])[0]\n",
    "            tmp_train_features = train_features[ind, :]\n",
    "            distances = np.array([euclidean_distance(train_sample, test_features) for train_sample in tmp_train_features])\n",
    "            print(distances.shape)\n",
    "            average_distances = np.mean(distances, axis=1)\n",
    "    \n",
    "            # Step 4: Find the overall average distance\n",
    "            overall_average_distance = np.mean(average_distances)\n",
    "            distance_dict[person_nums[ii]] = overall_average_distance\n",
    "    \n",
    "    \n",
    "    def normalize_to_weights(distance_dict):\n",
    "        distances = np.array(list(distance_dict.values()))\n",
    "        # Handle the case where a distance is zero to avoid division by zero\n",
    "        distances = np.clip(distances, a_min=1e-10, a_max=None)\n",
    "        weights = 1 / distances\n",
    "        normalized_weights = weights\n",
    "        # normalized_weights = weights / sum(weights)\n",
    "        # print(sum(weights))\n",
    "        # print(sum(normalized_weights))\n",
    "        # Assign the normalized weights back to the dictionary\n",
    "        normalized_weight_dict = dict(zip(distance_dict.keys(), normalized_weights))\n",
    "        return normalized_weight_dict\n",
    "    def scale_dict_values(my_dict):\n",
    "        scaled_dict = my_dict.copy()\n",
    "        min_val = min(scaled_dict.values())\n",
    "        max_val = max(scaled_dict.values())\n",
    "        \n",
    "        for key in scaled_dict:\n",
    "            scaled_dict[key] = 1 + 9 * (scaled_dict[key] - min_val) / (max_val - min_val)\n",
    "        \n",
    "        return scaled_dict\n",
    "    weights_dict = normalize_to_weights(distance_dict)\n",
    "    weights_dict = scale_dict_values(weights_dict)\n",
    "    print(weights_dict)\n",
    "    \n",
    "    w_train = np.zeros_like(p_train)\n",
    "    for i in range(len(w_train)):\n",
    "        w_train[i] = weights_dict[int(p_train[i])]\n",
    "    \n",
    "    w_train = np.squeeze(w_train)\n",
    "    \n",
    "    import sys\n",
    "    sys.path.append('..')\n",
    "    # for layer in model.layers[-4:]:\n",
    "    #     layer.trainable = False\n",
    "    # model = individual_model.individual_model(X_train_normalized)\n",
    "    model_name = './checkpoints/unknown_person_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=r, final_pruning_factor=r, start_epoch=1, end_epoch=20, frequency=1)\n",
    "    \n",
    "    history = model.fit(x=X_train_normalized_new, y=y_train,sample_weight= w_train,epochs=20, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])\n",
    "    import sys\n",
    "    sys.path.append('..')\n",
    "    # for layer in model.layers[-4:]:\n",
    "    #     layer.trainable = False\n",
    "    # model = individual_model.individual_model(X_train_normalized)\n",
    "    model_name = './checkpoints/unknown_person_baseline_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    model = tf.keras.models.load_model(model_name)\n",
    "    model_name = './checkpoints/unknown_person_baseline_2_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    idx += 1\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=r, final_pruning_factor=r, start_epoch=1, end_epoch=20, frequency=1)\n",
    "    \n",
    "    history = model.fit(x=X_train_normalized_new, y=y_train,epochs=20, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f71b59e4-6581-4ea8-994d-a02a6654ab6b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
